### PR TITLE
Use utf-8 to encode flyte deck

### PR DESCRIPTION
**Pull request by pingsutw**
# TL;DR
Use `utf-8` to encode flyte deck, and catch the error during generating the deck file to prevent the entire task from failing)
Fixes https://github.com/unionai/cloud/issues/3733

## Type
 - [x] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [ ] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
^^^

## Tracking Issue
_NA_

## Follow-up issue
_NA_

### CODE REVIEW

File: flytekit/deck/deck.py [hunk 1]
```diff
@@ -147,16 +146,19 @@ def _output_deck(task_name: str, new_user_params: ExecutionParameters):
...
+        logger.info(f"{task_name} task creates flyte deck html to file://{local_path}")
+        if ctx.execution_state.mode == ExecutionState.Mode.TASK_EXECUTION:
+            remote_path = f"{new_user_params.output_metadata_prefix}{os.sep}{DECK_FILE_NAME}"
+            kwargs: typing.Dict[str, str] = {
+                "ContentType": "text/html",  # For s3
+                "content_type": "text/html",  # For gcs
+            }
+            ctx.file_access.put_data(local_path, remote_path, **kwargs)
+    except Exception:
+        logger.error("Failed to write flyte deck html.")
```
Comment (wild-endeavor) [inline flytekit/deck/deck.py:161]: if we catch the exception we should print the original error

File: flytekit/deck/deck.py [hunk 2]
```diff
@@ -147,16 +146,19 @@ def _output_deck(task_name: str, new_user_params: ExecutionParameters):
...
-            "ContentType": "text/html",  # For s3
-            "content_type": "text/html",  # For gcs
-        }
-        ctx.file_access.put_data(local_path, remote_path, **kwargs)
+    try:
+        with open(local_path, "w", encoding="utf-8") as f:
+            f.write(_get_deck(new_user_params, ignore_jupyter=True))
+        logger.info(f"{task_name} task creates flyte deck html to file://{local_path}")
+        if ctx.execution_state.mode == ExecutionState.Mode.TASK_EXECUTION:
+            remote_path = f"{new_user_params.output_metadata_prefix}{os.sep}{DECK_FILE_NAME}"
```
Comment (wild-endeavor) [inline flytekit/deck/deck.py:154]: os.sep won't work on windows though.

Comment (pingsutw) [reply flytekit/deck/deck.py:154]: hmm, why? https://docs.python.org/3/library/os.html#os.sep

File: flytekit/deck/deck.py (hunk 1 above)
Comment (pingsutw) [reply flytekit/deck/deck.py]: updated it

File: flytekit/deck/deck.py (hunk 2 above)
Comment (wild-endeavor) [reply flytekit/deck/deck.py:154]: because this is the remote path right?  what if you're on windows?  this will return `\`, which s3 won't accept right?

Comment (pingsutw) [reply flytekit/deck/deck.py]: You're right. updated it, thanks!

**Review by eapolinario — APPROVED** [review]

**Pull request merged**